trite-osprey/src/List.js

```diff
@@ -0,0 +1,219 @@
+import React from 'react';
+import Cart from './Cart';
+
+class List extends React.Component{
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      beers : [],
+      currentBeer: null,
+      searchString: '',
+      visible: false,
+      active: this.props.active || 0,
+      beerList: [],
+      totalPrice: 0,
+      currentBeerObjects: []
+    }
+    this.handleClick = this.handleClick.bind(this)
+  }
+
+  componentWillMount() {
+    this.getAllBeers()
+  }
+
+  getAllBeers() {
+    fetch('http://localhost:5000/beers', {
+      method: 'get',
+    })
+    .then(response => response.json())
+    .then(results => {
+      this.setState({
+        beers: results
+      })
+    })
+  }
+
+  handleChange(event) {
+    this.setState( { searchString: event.target.value} )
+  }
+
+  hide() {
+    this.setState({visible: false})
+  }
+
+  handleClick(event, index) {
+    this.setState({visible: true, active: index, currentBeer: this.state.beers[index] })
+  }
+
+  addToCart() {
+    let { currentBeer, beerList, currentBeerObjects } = this.state
+    currentBeerObjects.push(currentBeer)
+    beerList.push(currentBeer.beername)
+
+    let priceArray = [];
```

Comment:
Putting the word `array` in a variable name should be avoided. Try something like this:

```js
let prices = currentBeerObjects.map(beer => beer.price)
```

Also it looks like you were using `Object.keys` as if `currentBeerObjects` was an object instead of an array. But when you initialized the `currentBeerObjects` in your `constructor` you set it to an array.